modules/effects/src/effect_creator.ts

```diff
@@ -5,10 +5,43 @@ import { EffectMetadata, EffectConfig } from './models';
 const CREATE_EFFECT_METADATA_KEY = '__@ngrx/effects_create__';
 
 type DispatchType<T> = T extends { dispatch: infer U } ? U : unknown;
+type ReturnType<T> = T extends false ? Observable<unknown> : Observable<Action>;
```

Comment:
Having this in the function definition prevented the documentation generation from the JSDocs.  Moved to an abstracted type.

Reply:
I think we use something more descriptive than `ReturnType` as it already exists in TypeScript.

Reply:
Changed to `ObservableReturnType` :+1: